.azure-pipelines/azure-pipelines-linux.yml

```diff
@@ -7,7 +7,7 @@ jobs:
     matrix:
       linux:
         CONFIG: azure-linux-64-comp7
-        IMAGE_NAME: condaforge/linux-anvil-comp7
+        IMAGE_NAME: condaforge/linux-anvil-cuda:9.2
```

Comment:
It would be good to not have to change this file in this PR

Reply:
Yes, we'll have to use docker on docker to do that. That would also help with testing CentOS7 packages that you wanted to add.

Reply:
This is just so we can test it here. We will revert it when we are done.

That said, we did start discussing how to do GPU tests in issue ( https://github.com/conda-forge/staged-recipes/issues/9773 ). Have a rough start of a potential solution in PR ( https://github.com/conda-forge/staged-recipes/pull/9804 ), but it could use help if anyone has time.

Reply:
I have a question: would it imply that all CUDA packages built this way for conda-forge, and the `- {{ compiler("cuda") }}` flag, will be pinned at CUDA 9.2?

Reply:
Right, this is why we need docs. 😅

So currently conda-forge builds against 3 CUDA versions. As some recipes have a CPU-only option, we support 1 CPU version + 3 CUDA versions in total. These can be seen [here]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/4797f214b86b0e7090d7f3f764368e277e0f3819/recipe/conda_build_config.yaml#L46-L50 ) (`None` is the CPU version). These versions map one-to-one to [these Docker images]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/4797f214b86b0e7090d7f3f764368e277e0f3819/recipe/conda_build_config.yaml#L107-L110 ) where the build occurs. Each package then gains a dependency on the [`cudatoolkit` package with the same major minor version]( https://github.com/conda-forge/nvcc-feedstock/blob/2097ca62eecccb3b3ac49b0e9b3c498bf8892734/recipe/meta.yaml#L19-L21 ).

Now what does this mean for CuPy. This means it will be built against all of the CUDA versions, but no CPU versions. Each build of CuPy will pin to a major minor version of CUDA (so 9.2, 10.0, and 10.1). Users can then constrain which build they get by selecting the version of `cudatoolkit` they want.

Reply:
Glad that helps. Thanks for the feedback. At some point it would be good if you can share your thoughts on what has been confusing and useful as well as what would be valuable to have in the docs. Have been collecting feedback from others in this doc issue ( https://github.com/conda-forge/conda-forge.github.io/issues/901 ). 🙂

Sorry we have not done a great job in publicizing it. Have been working through rough edges with individual contributors to try and smooth out the overall experience. That said, it would be useful to write up (even an incomplete) blogpost about the state of things from a high-level. Thanks for the nudge. 😉

FWICT the image is using CUDA 10.1 Update 2 (`nvcc` output below). Thanks for letting me know. IIUC that just affects `master` and the latest RC right (asked on the issue too)? In either case will follow that issue closely.

```
$ nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Sun_Jul_28_19:07:16_PDT_2019
Cuda compilation tools, release 10.1, V10.1.243
```

Good point. Have also been thinking about this. Could you please raise an issue about more detailed versioning on [the `nvcc` feedstock]( https://github.com/conda-forge/nvcc-feedstock )?

Reply:
OK will do. Also listing version strings here for future ref:
10.1: `10.1.105`
10.1 Update 1: `10.1.168`
10.1 Update 2: `10.1.243`

Reply:
See conda-forge/nvcc-feedstock#28.